content/references/protocol-reference/transactions/transaction-types/ammwithdraw.md

```diff
@@ -0,0 +1,116 @@
+---
+html: ammwithdraw.html
+parent: transaction-types.html
+blurb: Return LPTokens into an Automated Market Maker in exchange for a share of the assets the pool holds.
+labels:
+  - AMM
+status: not_enabled
+---
+# AMMWithdraw
+[[Source]](https://github.com/gregtatcam/rippled/blob/amm-core-functionality/src/ripple/app/tx/impl/AMMWithdraw.cpp "Source")
+<!-- TODO: Update source link to merged version when available -->
+
+{% include '_snippets/amm-disclaimer.md' %}
+
+Withdraw funds an Automated Market-Maker (AMM) instance by returning the AMM's liquidity provider tokens (LP Tokens).
+
+## Example {{currentpage.name}} JSON
+
+```json
+{
+    "Account" : "rJVUeRqDFNs2xqA7ncVE6ZoAhPUoaJJSQm",
+    "Amount" : {
+        "currency" : "TST",
+        "issuer" : "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd",
+        "value" : "5"
+    },
+    "Amount2" : "50000000",
+    "Asset" : {
+        "currency" : "TST",
+        "issuer" : "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd"
+    },
+    "Asset2" : {
+        "currency" : "XRP"
+    },
+    "Fee" : "10",
+    "Flags" : 1048576,
+    "Sequence" : 10,
+    "TransactionType" : "AMMWithdraw"
+}
+```
+
+{% include '_snippets/tx-fields-intro.md' %}
+
+| Field        | JSON Type           | [Internal Type][] | Required? | Description |
+|:-------------|:--------------------|:------------------|:----------|:------------|
+| `Asset`      | Object              | STIssue           | Yes       | The definition for one of the assets in the AMM's pool. In JSON, this is an object with `currency` and `issuer` fields (omit `issuer` for XRP). |
+| `Asset2`     | Object              | STIssue           | Yes       | The definition for the other asset in the AMM's pool. In JSON, this is an object with `currency` and `issuer` fields (omit `issuer` for XRP). |
+| `Amount`     | [Currency Amount][] | Amount            | No        | The amount of one asset to withdraw from the AMM. This must match the type of one of the assets (tokens or XRP) in the AMM's pool. |
+| `Amount2`    | [Currency Amount][] | Amount            | No        | The amount of another asset to withdraw from the AMM. If present, this must match the type of the other asset in the AMM's pool and cannot be the same type as `Amount`. |
+| `EPrice`     | [Currency Amount][] | Amount            | No        | The minimum effective price, in LP Token returned, to pay per unit of the asset to withdraw. |
```

Comment:
Is this definition of effective price correct? My understanding is that it is calculated as the ratio of the tokens swapped in vs out, giving the "effective price" of the trade. I don't think LP token quantity is a factor in the equation, but maybe i'm wrong and that works too

Reply:
Sorry, I'm wrong because of the txn type (withdraw, not swap). Ignore the above

Reply:
In the case of a withdraw with `EPrice`, what's swapped in _is_ LP Tokens